tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -2,78 +2,103 @@
 # See associated tribits/Copyright.txt file for copyright and license! #
 ########################################################################
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding HDF5.  These are used in case find_package(HDF5 ...) is not called
-# or does not find HDF5.  Also, these variables need to be non-null in order
-# to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
-
-set(REQUIRED_HEADERS hdf5.h)
-set(REQUIRED_LIBS_NAMES hdf5)
-
-if (HDF5_REQUIRE_FORTRAN)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_fortran)
-endif()
+set(HDF5_INTERNAL_IS_MODERN FALSE)
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} z)
-endif()
+if (Netcdf_ALLOW_MODERN)
+
+  set(minimum_modern_HDF5_version 1.13.2)
```

Comment:
@ibaned @bartlettroscoe 
Sorry for not noticing this prior to merging.  
What is needed in the HDF5-1.10.X and HDF5-1.12.X versions to make them usable for this.  Both of those versions are still in development, so it would be good to fix their CMake configurations if possible.

I can either contact the HDF5 group with the suggested changes, or one of you can also.  Note that we have priority support with HDF5, so they will listen and most likely make useful changes that we suggest.